src/Microsoft.TestPlatform.CommunicationUtilities/SocketClient.cs

```diff
@@ -65,22 +68,24 @@ public void Stop()
 
         private void OnServerConnected(Task connectAsyncTask)
         {
-            if (connectAsyncTask.IsFaulted)
-            {
-                throw connectAsyncTask.Exception;
-            }
-
-            this.channel = this.channelFactory(this.tcpClient.GetStream());
-            if (this.ServerConnected != null)
+            if (this.Connected != null)
             {
-                this.ServerConnected.SafeInvoke(this, new ConnectedEventArgs(this.channel), "SocketClient: ServerConnected");
+                if (connectAsyncTask.IsFaulted)
+                {
```

Comment:
Add EqtTrace on failure and success scenarios.

Reply:
Done